### PR TITLE
[cli] Use ensureLink() in `link` and `pull` commands

### PR DESCRIPTION
**Pull request by cb1kenobi**
The `link` and `pull` commands have essentially duplicate code from `ensureLink()`. This

Card: https://linear.app/vercel/issue/VCCLI-224/adopt-ensurelink-across-commands

The `dev` command should also be updated to use `ensureLink()`, however that may interfere with this in progress PR: https://github.com/vercel/vercel/pull/8666.

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [x] This PR has a concise title and thorough description useful to a reviewer
- [x] Issue from task tracker has a link to this PR

### CODE REVIEW

File: packages/cli/src/util/ensure-link.ts [hunk 1]
```diff
@@ -5,23 +5,46 @@ import param from './output/param';
...
+ * @param commandName - The name of the current command to print in the
+ * event of an error
+ * @param client - The Vercel Node.js client instance
+ * @param cwd - The current working directory
+ * @param yes - When `true`, skips questions when setting up new project
+ * @param opts.forceDelete - When `true`, deletes the project's `.vercel`
+ * directory
+ * @param opts.projectName - The project name to use when linking, otherwise
+ * the current directory
+ * @returns {LinkResult|number} Returns a numeric exit code when aborted or
```
Comment (EndangeredMassa) [inline packages/cli/src/util/ensure-link.ts]: **suggestion (soft-blocking):**

```suggestion
 * @returns {Promise<LinkResult|number>} Returns a numeric exit code when aborted or
```

This technically returns a `Promise<LinkResult | number>`.

In this or a follow-up PR: can we update this type?

**Review by EndangeredMassa — APPROVED** [review]

File: packages/cli/src/util/ensure-link.ts [hunk 2]
```diff
@@ -5,23 +5,46 @@ import param from './output/param';
 import { getCommandName } from './pkg-name';
```
Comment (TooTallNate) [inline packages/cli/src/util/ensure-link.ts]: While we're at it, let's move this file into `util/link`?

**Review by TooTallNate — APPROVED** [review]

File: packages/cli/src/commands/link/index.ts [hunk 3]
```diff
@@ -70,31 +68,14 @@ export default async function main(client: Client) {
   }
 
   const cwd = argv._[1] || process.cwd();
-  const link = await setupAndLink(client, cwd, {
+
+  const link = await ensureLink('link', client, cwd, !!argv['--yes'], {
     forceDelete: true,
-    autoConfirm: argv['--yes'],
     projectName: argv['--project'],
-    successEmoji: 'success',
```
Comment (styfle) [inline packages/cli/src/commands/link/index.ts:77]: I think we still need successEmoji

File: packages/cli/src/commands/pull.ts [hunk 4]
```diff
@@ -83,43 +77,6 @@ function parseArgs(client: Client) {
...
-async function ensureLink(
-  client: Client,
-  cwd: string,
-  yes: boolean
-): Promise<LinkResult | number> {
-  let link = await getLinkedProject(client, cwd);
-  if (link.status === 'not_linked') {
-    link = await setupAndLink(client, cwd, {
-      autoConfirm: yes,
-      successEmoji: 'link',
```
Comment (styfle) [inline packages/cli/src/commands/pull.ts:99]: Same here: it’s a different successEmoji

File: packages/cli/src/util/ensure-link.ts [hunk 5]
```diff
@@ -5,23 +5,46 @@ import param from './output/param';
...
   client: Client,
   cwd: string,
-  yes: boolean
+  yes: boolean,
+  opts?: LinkOpts
 ): Promise<LinkResult | number> {
   let link = await getLinkedProject(client, cwd);
   if (link.status === 'not_linked') {
     link = await setupAndLink(client, cwd, {
       autoConfirm: yes,
```
Comment (styfle) [inline packages/cli/src/util/ensure-link.ts]: Let’s pass autocomfirm and successEmoji in opts. And I assume setupMsg should also be in opts. Not sure why these are all hardcoded

**Review by styfle — CHANGES_REQUESTED** [review]
Looks like a bug since `successEmoji` is missing

File: packages/cli/src/commands/pull.ts (hunk 4 above)
Comment (cb1kenobi) [reply packages/cli/src/commands/pull.ts:99]: `successEmoji` now defaults to `'link'`.

**Comment by cb1kenobi** [comment]
I believe all issues have been address. Please re-review if you have a moment. Thanks!

**Review by EndangeredMassa — APPROVED** [review]

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**